dns/tsigkeyring.py

```diff
@@ -55,5 +55,10 @@ def b64encode(secret):
         if isinstance(key, bytes):
             textring[name] = b64encode(key)
         else:
-            textring[name] = (key.algorithm.to_text(), b64encode(key.secret))
+            if isinstance(key.secret, bytes):
+                text_secret = b64encode(key.secret)
+            else:
+                text_secret = str(key.secret)
```

Comment:
What would the secret be, in this case?  There aren't any corresponding changes in `from_text`, so I guess not something reversible?

Reply:
Right - it's not reversible.  The repr of the key.secret is the gssapi type (although in this case, it's the mock):

`{'keyname.': 'NjHwPsMKjdN++dOfE5iAiQ==', 'gsstsigtest.': ('gss-tsig.', "<Mock id='140055329354640'>")}`